pkg/indexer/index.go

```diff
@@ -0,0 +1,168 @@
+package indexer
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/marcus-crane/khinsider/v2/pkg/update"
+	"github.com/pterm/pterm"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/marcus-crane/khinsider/v2/pkg/scrape"
+	"github.com/marcus-crane/khinsider/v2/pkg/types"
+	"github.com/marcus-crane/khinsider/v2/pkg/util"
+)
+
+const (
+	LocalIndex  = "index.json"
+	RemoteIndex = "https://raw.githubusercontent.com/marcus-crane/khinsider-index/main/index.json"
+)
+
+func getConfigPath(filename string) string {
+	usr, err := user.Current()
+	if err != nil {
+		panic(err)
+	}
+	return filepath.Join(usr.HomeDir, ".khinsider", filename)
```

Comment:
Yeah, that's a good point. I'm mixing two things here since eventually I was thinking about a config file but that should live elsewhere. I still need to review the XDG spec in detail.

Reply:
Done